README.md

```diff
@@ -11,12 +11,12 @@ Prop supports two limiting strategies:
 To get going with Prop, you first define the read and write operations. These define how you write a registered request and how to read the number of requests for a given action. For example, do something like the below in a Rails initializer:
 
 ```ruby
-Prop.read do |key|
-  Rails.cache.read(key)
-end
-
-Prop.write do |key, value|
-  Rails.cache.write(key, value)
+# TODO Prop.cache = Rails.cache ... and deprecate old write / read
+Prop.read { |key| Rails.cache.read(key) }
+Prop.write { |key, value| Rails.cache.write(key, value) }
+Prop.increment do |key, value| 
+  # alternatively implement this with write(read + value), but you will suffer from race conditions
+  Rails.cache.increment(key, value) || Rails.cache.write(key, value)
```

Comment:
I removed that and converted to Prop.cache = Rails.cache ... users can
assign whatever cache they want and we check that it supports the correct
methods

On Tue, Oct 20, 2015 at 5:12 PM, Gary Grossman <notifications@github.com>
wrote:

> In README.md <https://github.com/zendesk/prop/pull/7#discussion_r42570975>
> :
>
> > @@ -11,12 +11,12 @@ Prop supports two limiting strategies:
> >  To get going with Prop, you first define the read and write operations. These define how you write a registered request and how to read the number of requests for a given action. For example, do something like the below in a Rails initializer:
> >
> >  ```ruby
> > -Prop.read do |key|
> > -  Rails.cache.read(key)
> > -end
> > -
> > -Prop.write do |key, value|
> > -  Rails.cache.write(key, value)
> > +# TODO Prop.cache = Rails.cache ... and deprecate old write / read
> > +Prop.read { |key| Rails.cache.read(key) }
> > +Prop.write { |key, value| Rails.cache.write(key, value) }
> > +Prop.increment do |key, value|
> > +  # alternatively implement this with write(read + value), but you will suffer from race conditions
> > +  Rails.cache.increment(key, value) || Rails.cache.write(key, value)
>
> So this makes prop dependent on rails? Do we want that?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/prop/pull/7/files#r42570975>.
>